src/modal-template.ts

```diff
@@ -0,0 +1,244 @@
+          <header style="background-color: ${this.config.headerColor}">
+            ${this.config.showCloseButton ? this.closeButtonTemplate : ''}
+            <div class="logo-icon">
+              <img
+                src="https://archive.org/images/logo_arches.png"
+                alt="Internet Archive Logo"
+              />
+            </div>
+            ${this.config.title
+              ? html`<h1 class="title">${this.config.title}</h1>`
+              : ''}
+            ${this.config.subtitle
+              ? html`<h2 class="subtitle">${this.config.subtitle}</h2>`
+              : ''}
+          </header>
+          <section class="modal-body">
+            <div class="content">
+              <div
+                class="processing-logo ${this.config.showProcessingIndicator
+                  ? ''
+                  : 'hidden'}"
+              >
+                <ia-activity-indicator
+                  .mode=${this.config.processingImageMode}
+                ></ia-activity-indicator>
+              </div>
+
+              ${this.config.headline
+                ? html` <h1 class="headline">${this.config.headline}</h1> `
+                : ''}
+              ${this.config.message
+                ? html` <p class="message">${this.config.message}</p> `
+                : ''}
+
+              <div class="slot-container">
+                <slot> </slot>
+              </div>
+            </div>
+          </section>
+        </div>
+      </div>
+    `;
+  }
+
+  /**
+   * Dispatch the `closeButtonPressed` event to the consumer
+   *
+   * @private
+   * @memberof ModalTemplate
+   */
+  private handleCloseButton(): void {
+    const event = new Event('closeButtonPressed');
+    this.dispatchEvent(event);
+  }
+
+  /**
+   * The close button template
+   *
+   * @readonly
+   * @private
+   * @type {TemplateResult}
+   * @memberof ModalTemplate
+   */
+  private get closeButtonTemplate(): TemplateResult {
+    return html`
+      <button
+        type="button"
+        class="close-button"
+        @click=${this.handleCloseButton}
+      >
+        ${closeIcon}
+      </button>
+    `;
+  }
+
+  /** @inheritdoc */
+  static get styles(): CSSResult {
+    const processingImageSize = css`var(--processingImageSize, 75px)`;
+
+    const modalCornerRadius = css`var(--modalCornerRadius, 10px)`;
+    const modalBorder = css`var(--modalBorder, 2px solid black)`;
+    // if the content of the modal is too big to fit on screen, this sets the bottom margin
+    // it's not exact, but a close estimation
+    const modalBottomMarginCss = css`var(--modalBottomMargin, 25px)`;
+    const modalTopMarginCss = css`var(--modalTopMargin, 50px)`;
+    const modalHeaderBottomPaddingCss = css`var(--modalHeaderBottomPadding, 0.5em)`;
+
+    const modalBottomPadding = css`var(--modalBottomPadding, 20px)`;
+    const scrollOffset = css`var(--modalScrollOffset, 5px)`;
+
+    const titleFontSize = css`var(--modalTitleFontSize, 18px)`;
```

Comment:
We've been using rem units for font sizes so they scale proportionally when the root element's font size changes. Like with Petabox, I like to start with the html element set to 10px, then all font sizes in rem are easy to determine. Then if we want to alter all font sizes proportionally based on viewport size, for example, we're all set.

Reply:
I tried setting these to use `rem` values, but they don't seem to change depending on the base font size. They always seem to be relative to the base font size of that type. ie `titleFontSize` is on an `h1`, which has a browser default of `16px`. Setting this to `1.4rem` became around `22px` instead of, say, `14px` for a `10px` base. I'm not sure if I'm setting that correctly or if that's intentional with the Shadow DOM.

Reply:
In your demo HTML file, you need to set the font size to 10px.

```css
html {
  font-size: 10px;
}
```

The topnav does this to ensure parity between demo and Petabox integration.

Reply:
Ah I had it set on the `body`. Now it's working.